inst/include/dust/gpu/launch_control.hpp

```diff
@@ -101,8 +101,8 @@ inline launch_control launch_control_model(size_t n_particles,
 
   const size_t n_pars_effective = n_particles / n_particles_each;
   const int warp_size = dust::gpu::warp_size;
-  const size_t warp_block_size =
-    warp_size * (n_particles_each + warp_size - 1) / warp_size;
+  const size_t warp_block_count = (n_particles_each + warp_size - 1) / warp_size;
+  const size_t warp_block_size = warp_size * warp_block_count;
```

Comment:
there was an order of operations issue here; for integers `a * b / c` is not always the same as `a * (b / c)` if `b` is not a multiple of `c`